foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/ConversionManager.java

```diff
@@ -169,8 +171,10 @@ public Object clone() {
                 return null;
             }
         }
-
-        if ((sourceObject.getClass() == javaClass) || (javaClass == null) || (javaClass == ClassConstants.OBJECT) || (javaClass == ClassConstants.BLOB) || (javaClass == ClassConstants.CLOB)) {
+        if (sourceObject.getClass() == javaClass || javaClass == null || javaClass == ClassConstants.OBJECT
+                || javaClass == ClassConstants.BLOB || javaClass == ClassConstants.CLOB
+                // JSON has its own default converter registered
+                || javaClass == ClassConstants.JSON_VALUE || javaClass == ClassConstants.JSON_ARRAY || javaClass == ClassConstants.JSON_OBJECT) {
             return (T) sourceObject;
```

Comment:
this is the place where the conversion should happen with no need for additional methods in the dbplatform

Reply:
Well, that's not that easy. Conversion will be platform specific for some databases, e.g. Oracle 21c:
https://docs.oracle.com/en/database/oracle/oracle-database/21/jajdb/oracle/sql/json/package-summary.html
```
OracleJsonObject pear = rs.getObject(1, OracleJsonObject.class);
```
All Oracle specific JSON types implement javax.json.JsonValue interface so I have to overwrite both - ResultSet retrieval and conversion code in new Oracle21cPlatform class.